src/staking/stakes_db.cpp

```diff
@@ -426,12 +444,71 @@ const AmountByPeriodArray& CStakesDBCache::getAmountsByPeriods() const {
 }
 
 // This method assumes that nothing else from stake reward is modified. Checks are not performed because they would need
-// DB lookups, which are too expensive. If
+// DB lookups, which are too expensive.
 bool CStakesDBCache::updateStakeEntry(const CStakesDbEntry &entry) {
     if (m_viewonly) {
         LogPrintf("ERROR: Cannot modify a viewonly cache");
         return false;
     }
     m_stakes_map[entry.getKey()] = entry;
     return true;
-}
\ No newline at end of file
+}
+
+CFreeTxInfo CStakesDBCache::getFreeTxInfoForScript(const CScript &script) const {
+    if (m_viewonly) {
+        return m_base_db->getFreeTxInfoForScript(script);
```

Comment:
I do not see big difference between this function and body of the getFreeTxInfoForScript. In ideal world this logic should be abstracted into some kind of container class. Or algorithm/template function.

Reply:
I'll do a common interface for `CStakesDB` and `CStakesDBCache` if we have some time left :)